### PR TITLE
Gatsby setup

### PR DESCRIPTION
**Pull request by schickm**
So here's an initial "Proof of concept" of the idea of pulling in content from various working groups and displaying it on a single site:

https://ei2030.github.io/EI2030/

I've grouped the work in meaningful commits, so feel free to browse by commit for better context.  I'll annotate the various files in this PR to provide a little better explanation of what's going on here.

The PR includes an addition of a deployment script (that's what I've used to deploy to the gh pages url above).  Moving forward we might want to deploy to a "staging" site that is prefixed by the branch name, and then only deploy to the main GH pages site when something merges to the `main` branch.  I'll ticket this up as a future todo

### CODE REVIEW

File: src/data/working-groups.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[
+  {
+    "name": "Example A",
+    "gitRemote": "https://github.com/EI2030/EI2030-example-working-group-a.git"
+  },
+  {
+    "name": "Group B",
+    "gitRemote": "https://github.com/EI2030/EI2030-example-working-group-b.git"
+  }
+]
```
Comment (schickm) [inline src/data/working-groups.json:10]: This is the entry point where we would add a new working group to the site.  All we should need to do is add it once here, and all markdown should be pulled in.

File: gatsby-config.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    },
+    ...workingGroups.map(({ name, gitRemote }) => ({
+      resolve: "gatsby-source-git",
+      options: {
+        name,
+        branch: "main",
+        patterns: "website/**",
+        remote: gitRemote,
+      },
+    })),
```
Comment (schickm) [inline gatsby-config.js:48]: Here we are using the contents of the `working-groups.json` to dynamically define our "git sources".  You'll note that there are certain properties that are statically defined like `patterns: "website/**"`.  This informs the plugin what subdirectory to go looking for content inside of.  

So the assumption here is that all working group repositories will have a subdirectory named `website/` and that we'll pull in all markdown from there.

Comment (alex-a-soto) [reply gatsby-config.js:48]: @schickm This looks great. I like the use of the `working-groups.json` to define our git sources, the properties defining the branch, pattern, and remote.

File: src/data/working-groups.json (hunk 1 above)
Comment (alex-a-soto) [reply src/data/working-groups.json:10]: Great thank you! :slightly_smiling_face:

**Review by alex-a-soto — APPROVED** [review]
@schickm Overall, everything looks great. The initial Gatsby setup is a good start for the website's main functionality to aggregate from different repositories / working groups.

Having a "staging area," as you mentioned in your earlier comment, is a good next step.

Thank you for your contributions and support.

**Pull request merged**